### PR TITLE
feat(ui): Added breadcrumb's level

### PR DESCRIPTION
**Pull request by priscilawebdev**
This feature is under the feature flag `breadcrumbs-v2`

**Description:**

- Added the level of breadcrumb
- Added breadcrumbs Header
- Refactored the detailed information that is displayed

**Next steps:**
- Add level in the filter
- Make all filter's option not checked by default
- Finish Summary implementation
- Improve full text search
- Create Filter tests
- Improve Mobile experience 

closes: https://app.asana.com/0/1158284503473033/1170567727546273

The breadcrumbs section in this PR, looks like:

image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataException.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  const {data} = breadcrumb;
+  return (
+    <BreadcrumbDataSummary kvData={omit(data, ['type', 'value'])}>
+      {data?.type &&
+        getBreadcrumbCustomRendererValue({
+          value: <strong>{`${data.type}: `}</strong>,
+          meta: getMeta(data, 'type'),
+        })}
+      {defined(data) &&
+        defined(data?.value) &&
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataException.tsx:25]: Should we merge this?

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataHttp.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          value: <strong>{`${data.method} `}</strong>,
+          meta: getMeta(data, 'method'),
+        })}
+      {data?.url &&
+        getBreadcrumbCustomRendererValue({
+          value: renderUrl(data.url),
+          meta: getMeta(data, 'url'),
+        })}
+      {defined(data) &&
+        defined(data.status_code) &&
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataHttp.tsx:51]: Let's merge this too

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {getMeta} from 'app/components/events/meta/metaProxy';
+import space from 'app/styles/space';
+import {defined} from 'app/utils';
+
+import getBreadcrumbCustomRendererValue from '../../breadcrumbs/getBreadcrumbCustomRendererValue';
+
+type KvData = {
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx:10]: Should this use Record?

**Review by matejminar — COMMENTED** [review]
Good job 👏

Do we need the header / is the design final? I tested locally and it looks kinda heavy to me.

Could you please do a Percy snapshot with `breadcrumbs-v2` feature enabled so that we can track these changes more easily?

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataException.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataException.tsx:25]: I have updated it...I believe now it's better...but the reason I did like above is that `data` can be `undefined` or `null` and `value` can be `0 `or something else

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx:10]: sure..thanks for the tip 😉

**Comment by priscilawebdev** [comment]
@matejminar thanks for the nice feedbacks 🙃 I will have a meeting with Dora today to talk about the new breadcrumb's design....we might remove the header, but first I would like to have some more internal feedbacks... 

About Percy ... I will find out how I can add the feature flag 😉

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import Tooltip from 'app/components/tooltip';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+
+type Props = {
+  category?: string;
+};
+
+const BreadcrumbCategory = ({category = 'generic'}: Props) => (
+  <div>
+    <Tooltip title={category}>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx:13]: Is this tooltip doing much if it is repeating the content that is already displayed?

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataHttp.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        </ExternalLink>
+      ) : (
+        <span>{url}</span>
+      );
+    }
+
+    try {
+      return JSON.stringify(url);
+    } catch {
+      return t('Invalid URL');
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataHttp.tsx:34]: Should we emit the original value here instead of an error message?

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        </div>
+        {this.renderData()}
+      </div>
+    );
+  }
+}
+
+export default BreadcrumbDataSummary;
+
+const StyledPre = styled('pre')`
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx:76]: I feel like I've seen this component name/CSS before in this pull request should there be a reusable component for this?

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbsListHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  border-bottom: 1px solid ${p => p.theme.borderDark};
+  background: ${p => p.theme.offWhite};
+`;
+
+const BreadcrumbListHeaderItem = styled('div')`
+  color: ${p => p.theme.gray3};
+  font-size: ${p => p.theme.fontSizeSmall};
+  font-weight: 600;
+  text-transform: uppercase;
+  border-radius: 4px 4px 0 0;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbsListHeader.tsx:34]: Should this be `theme.borderRadius`?

File: tests/acceptance/test_issue_details.py [hunk 8]
```diff
@@ -15,6 +15,9 @@
 
 
 class IssueDetailsTest(AcceptanceTestCase, SnubaTestCase):
+    def __init__(self, features):
+        self.features = features
```
Comment (markstory) [inline tests/acceptance/test_issue_details.py:19]: Why is this needed now?

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+import Tooltip from 'app/components/tooltip';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+
+type Props = {
+  category?: string;
+};
+
+const BreadcrumbCategory = ({category = 'generic'}: Props) => (
+  <div>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx:12]: Is this `div` needed?

File: tests/acceptance/test_issue_details.py (hunk 8 above)
Comment (priscilawebdev) [reply tests/acceptance/test_issue_details.py]: I was just trying something ... I've updated this code ;-)

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx (hunk 9 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx:12]: unfortunately yes, otherwise the tooltip will not be aligned to the center

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx (hunk 6 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx:76]: I've changed in the current breadcrumb and in the new version..that is the reason you have seen this twice... in the future it's gonna be only one ..so no need to make it reusable 😉

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx (hunk 4 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbCategory.tsx:13]: yes, the tooltip is useful, because sometimes the content is too long and we display ellipsis

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataHttp.tsx (hunk 5 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataHttp.tsx:34]: I am not sure (I have to check), but I have the feeling that if we return the original value here, we might cause some error

File: tests/acceptance/test_issue_details.py [hunk 10]
```diff
@@ -79,11 +79,24 @@ def test_cocoa_event(self):
         self.page.visit_issue(self.org.slug, event.group.id)
         self.browser.snapshot("issue details cocoa")
 
+    def test_cocoa_event_breadcrumb_v2(self):
+        with self.feature("organizations:breadcrumbs-v2"):
+            # TODO: This should become its own "rust" platform type
+            event = self.create_sample_event(platform="native", sample_name="Rust")
```
Comment (markstory) [inline tests/acceptance/test_issue_details.py:85]: Should this be cocoa?

**Review by markstory — APPROVED** [review]

File: tests/acceptance/test_issue_details.py (hunk 10 above)
Comment (priscilawebdev) [reply tests/acceptance/test_issue_details.py:85]: sure thing :) I will fix in another PR ...promise 🤞
